docs/end-user/filtering-call-trees.md

```diff
@@ -0,0 +1,81 @@
+* After searching: [https://perfht.ml/2rbcj0N](https://perfht.ml/2rbcj0N)
+
+The following are some ideas on how to use search terms:
+
+ * `js::` - Filter for a C++ namespace.
+ * `www.example.com` - Filter for a domain name
+ * `www.example.com/assets/scripts.js` - Filter for a single script.
+
+## Implementation filter
+
+![A screenshot of the implementation filter dropdown.](../images/implementation-2018-05.jpg)
+
+The implementation filter is useful for narrowing down the call tree to a subset of stack types. For instance it can be useful to only show native stacks (C++ and Rust) or to only show JavaScript stacks.
+
+The following graphic demonstrates how it can be useful to filter to only JavaScript stacks.
+
+![](../images/implementation-filter.svg)
+
+Before the the implementation filter, the call tree contains both C++ JavaScript internals, as well as the JavaScript stacks. This interleaving could be useful when diagnosing how the JS engine is optimizing code, but for normal JS performance profiling this could be confusing to find hot JS functions. When the native stacks are removed, the call tree that is generated is much more sensible for representing the execution of JavaScript code.
+
+The implementation filter will modify the shape of stacks, and generate a completely new call tree. If any of the filtered stacks are empty, then those samples will be dropped from the analysis.
+
+## Invert call stack
+
+![A screenshot of the "Invert call stacks" checkbox.](../images/invert-2018-05.jpg)
+
+Inverting the call stacks of samples will produce an entirely new call tree. All of the self time will be at the roots of the call tree. The implications to the shape of the call tree may be surprising, but consider the following graphics.
+
+![An image showing samples on the left, and the call tree on the right. This is before the invert operation.](../images/invert-before.svg)
+
+In this profile, for the sake of brevity, there are only 3 samples that were collected 1ms apart. In most uninverted call trees, there is only one root node. In this case `A` calls `B`, and it's not until the third stack frames where the calls diverge to function `X` and function `C`. The call tree on the right is what is produced from these samples. Now consider inverting all of the stacks in the samples.
+
+![An image showing samples on the left, and the call tree on the right. This is after the invert operation.](../images/invert-after.svg)
+
+Previously, the stacks `X -> Y -> Z` were each at different levels. However, in the inverted view, the stacks are now at the same level. When the call tree is created from these samples, there are now two different roots, starting from the function `Z` and the function `E`. The inversion made the `Z -> Y -> X` portions of the stacks fall into the same level.
+
+Inversion is most useful in surfacing the functions where time is actually being spent. The trees are typically much larger and noisier than uninverted call trees.
```

Comment:
For a later patch, I miss these informations about inverted call trees:

1. I miss a sentence explaining again that the call tree is sorted, and so the functions where we spend the most time are at the top.

2. Also, it's very common to have samples like this:
A -> B -> C
then
A -> B -> C -> D
especially when C has some self time.
In uninverted call trees, this isn't very easy to spot without squinting the eyes, but inverted call trees make it easy to spot as we'll see both C and D at the top level.
You have such an example in the "call tree" document that you could possibly reuse here.

I also don't like the word "noisier" so much, it has a negative connotation and means that inverted call trees are difficult to make sense of. Maybe that's not what you intend here, but this is how I understand it.
(note: I don't want to argue about it before landing the PR, I'm fine landing the PR with this word, but let's discuss about it later :) )